spec/01-unit/27-queue_spec.lua

```diff
@@ -44,23 +42,49 @@ describe("plugin queue", function()
     ngx.ctx.workspace = nil
   end)
 
+  local unmock
+  local now_offset = 0
+  local log_messages
+
   before_each(function()
-    old_log = kong.log
+    local real_now = ngx.now
+    now_offset = 0
     log_messages = ""
     local function log(level, message) -- luacheck: ignore
       log_messages = log_messages .. level .. " " .. message .. "\n"
     end
-    kong.log = {
-      debug = function(message) return log('DEBUG', message) end,
-      info = function(message) return log('INFO', message) end,
-      warn = function(message) return log('WARN', message) end,
-      err = function(message) return log('ERR', message) end,
-    }
-  end)
 
-  after_each(function()
-    kong.log = old_log -- luacheck: ignore
+    mocker.setup(function(f)
+      unmock = f
+    end, {
+      kong = {
+        log = {
+          debug = function(message) return log('DEBUG', message) end,
+          info = function(message) return log('INFO', message) end,
+          warn = function(message) return log('WARN', message) end,
+          err = function(message) return log('ERR', message) end,
+        }
+      },
+      ngx = {
+        ctx = {
+          -- make sure our workspace is nil to begin with to prevent leakage from
+          -- other tests
+          workspace = nil
+        },
+        -- We want to be able to fake the time returned by ngx.now() only in the queue module and leave everything
+        -- else alone so that we can see what effects changing the system time has on queues.
+        now = function()
+          local called_from = debug.getinfo(2, "nSl")
+          if string.match(called_from.short_src, "/queue.lua$") then
+            return real_now() + now_offset
+          else
+            return real_now()
```

Comment:
This _appears_ to be unnecessary. Monkeypatching a function should only affect the current scope of the test, let alone of the system.
If this way of isolating a module from getting patched, we should find a more generic way to alter system internals.

Reply:
The problem is that `wait_until` and other testing utilities don't like the time being manipulated under them.  If there is a better way to manipulate the time only for the code under test, not the testing environment, I'm open to suggestions.

Reply:
I don't have a good solution either. Lets keep this on our radar.